grails-app/services/org/pih/warehouse/replenishment/ReplenishmentService.groovy

```diff
@@ -162,9 +162,15 @@ class ReplenishmentService {
         return order
     }
 
-    void validateRequirement(ReplenishmentItem item) {
-        if (item.quantity > productAvailabilityService.getQuantityAvailableToPromiseByProductNotInBin(item.location, item.binLocation, item.product)) {
-            throw new ValidationException("There is not available that quantity of the product with id: ${item.product.id}")
+    void validateRequirement(ReplenishmentItem item, Boolean updating = false) {
+        if (updating) {
+            if (item.quantity > productAvailabilityService.getQuantityAvailableToPromiseByProductNotInBin(item.location, item.binLocation, item.product) + item.quantity) {
```

Comment:
The `+ item.quantity` should rather be a sum of quantity from item.picklistItems and make it a bit more readable:
```
def quantityAvailableToPromise = productAvailabilityService.getQuantityAvailableToPromiseByProductNotInBin(item.location, item.binLocation, item.product)
def quantityPicked = sum of quantitiy from item.picklistItems
def quantityAvailableToPromiseWithPicked = quantityAvailableToPromise + quantityPicked
```